### ISSUE TITLE
Build of libsolv failed with SWIG 4.0.0

### ISSUE THREAD

**Issue by jplesnik** [openSUSE/libsolv#325]
Rebuild of libsolv package in Fedora with SWIG 4.0.0 failed with following error.

```bindings/python/solv_python.c: In function '_wrap_Pool_free':
bindings/python/solv_python.c:11863:21: error: 'obj0' undeclared (first use in this function); did you mean 'j0'?
11863 |     SWIG_ConvertPtr(obj0, &argp1,SWIGTYPE_p_Pool, SWIG_POINTER_DISOWN |  0 );
      |                     ^~~~
```

I wrote about the error to [SWIG upstream](https://sourceforge.net/p/swig/mailman/swig-devel/thread/61396b43-59bd-e7e7-dd65-c77985818f51%40redhat.com/#msg36590176). You can find more info about the issue and solution in the mails.

The fix, which works for Fedora, is attached to [Fedora BZ#1707367](https://bugzilla.redhat.com/show_bug.cgi?id=1707367)

**Comment by Conan-Kudo** [comment]
@jplesnik This is great, but could you please send the patch in the form of a pull request?

**Comment by mlschroe** [comment]
Merged as commit 78e2076be0969bddf599f805985ae7bd39298c62

Thanks!
